### ISSUE TITLE
warn when unresolved symbols are present in namespace metadata

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#602]
Linting a custom macro as `->` doesn't work when not quoting the `:clj-kondo/config` metadata.
In that case clj-kondo should warn about unresolved symbols (in the same way as we fixed #597).

E.g.: `(ns foo {:clj-kondo/config {:lint-as {foo bar}}})` => unresolved symbols foo and bar.

This however works:
``` clojure
(ns lint-as--GT
  {:clj-kondo/config '{:lint-as {lint-as--GT/injecting-thread-first clojure.core/->}}})

(defmacro injecting-thread-first
  [& body]
  `(-> :inject ~@body))

(defn arity-2 [x y]
  [x y])

(-> (arity-2 :y :z)
    (arity-2 :foo))

(injecting-thread-first (arity-2 :y :z)
                        (arity-2 :foo))
```

This gives:

```
$ clj-kondo --lint /tmp/johan.clj
linting took 12ms, errors: 0, warnings: 0
```
